src/chainstate/stacks/db/blocks.rs

```diff
@@ -4907,6 +4907,15 @@ impl StacksChainState {
                             receipts.append(&mut clarity_tx.block.initialize_epoch_2_4()?);
                             applied = true;
                         }
+                        StacksEpochId::Epoch25 => {
```

Comment:
Block-processing and block-decoding needs to be epoch-gated.  A block should not even _decode_ if it has a transaction with the new multisig spending condition.  This is because node without this PR wouldn't be able to decode the block either, and nodes with this patch must behave _exactly_ like the current mainnet nodes up until epoch 3.0 activates.

Reply:
@jcnelson @jbencin one question about this comment
Won't it be enough to add the gating logic to the `process_transaction` and `can_include_tx` methods?
The logic from the first one is used anyway in `process_block_transactions` and `process_microblocks_transactions` so blocks won't pass this check, and the second one is used for mempool transaction validation - added there the gating logic too

Or did you mean adding it also to block deserialization methods? These methods will need to have more dependencies to get the current epoch to conduct this verification, which is probably not appropriate for standard methods like `consensus_deserialize`